### PR TITLE
Refine LLM getting started guide for uniformity, fix critical errors

### PR DESCRIPTION
**Pull request by GregoryComer**
Update the LLM getting started guide for uniform tone and tense. Informally following the Google developer documentation style guide: https://developers.google.com/style. Also, resolve a number of outstanding issues with incorrect or misleading documentation and steps.

For reference, here are links to the current and proposed LLM guide:
https://docs-preview.pytorch.org/pytorch/executorch/2911/llm/getting-started.html (proposed)
https://pytorch.org/executorch/main/llm/getting-started.html (live)

### CODE REVIEW

File: install_requirements.sh [hunk 1]
```diff
@@ -9,14 +9,22 @@
 # Dependencies are defined in .pyproject.toml
 if [[ -z $PYTHON_EXECUTABLE ]];
 then
-  if [[ -z $CONDA_DEFAULT_ENV ]] || [[ $CONDA_DEFAULT_ENV == "base" ]];
+  if [[ -z $CONDA_DEFAULT_ENV ]] || [[ $CONDA_DEFAULT_ENV == "base" ]] || [[ ! -x "$(command -v python)" ]];
```
Comment (GregoryComer) [inline install_requirements.sh:12]: Not sure how this happened, but heard of a case where python wasn't on the system path while in a conda env. Something to do with env/shell conflicts on mac? This change checks to make sure python actually exists.

File: install_requirements.sh [hunk 2]
```diff
@@ -9,14 +9,22 @@
...
     PYTHON_EXECUTABLE=python
   fi
 fi
 
+if [[ "$PYTHON_EXECUTABLE" == "python" ]];
+then
+  PIP_EXECUTABLE=pip
+else
+  PIP_EXECUTABLE=pip3
+fi
```
Comment (GregoryComer) [inline install_requirements.sh:25]: On mac when not under conda, pip may not exist. We should use pip3 when PYTHON_EXECUTABLE=python3.

File: docs/source/llm/getting-started.md [hunk 3]
```diff
@@ -14,221 +14,308 @@
 
 ## Prerequisites
 
-Let’s start by getting an ExecuTorch environment:
```
Comment (byjlw) [inline docs/source/llm/getting-started.md:17]: this is better. I like not having external links for instructions.

File: docs/source/llm/getting-started.md [hunk 4]
```diff
@@ -14,221 +14,308 @@
...
 
-1.  Create a third-party folder (Keeps the file paths organized)
-```
-mkdir  third-party
-cd  third-party
-```
-2. If you’re new to ExecuTorch follow [these steps](https://pytorch.org/executorch/main/getting-started-setup.html#set-up-your-environment) to set up your environment.
+::::{tab-set}
+:::{tab-item} conda
+Instructions on installing miniconda can be [found here](https://docs.anaconda.com/free/miniconda).
```
Comment (byjlw) [inline docs/source/llm/getting-started.md:23]: can we eliminate the need for Conda. 
Can you re-write this so that it uses env?

Comment (byjlw) [reply docs/source/llm/getting-started.md:23]: O looks like you have both options here. Perfect. I'd change the instructions just a tad so it calls out two different options up front before giving the conda instructions

File: docs/source/llm/getting-started.md [hunk 5]
```diff
@@ -14,221 +14,308 @@
...
+        // Convert the output logits from EValue to std::vector, which is what
+        // the sampler expects.
+        Tensor logits_tensor = logits_evalue.get()[0].toTensor();
+        std::vector<float> logits(logits_tensor.data_ptr<float>(), 
+            logits_tensor.data_ptr<float>() + logits_tensor.numel());
 
-```
-string  prompt  =  "Hello world!";
-```
+        // Sample the next token from the logits.
```
Comment (byjlw) [inline docs/source/llm/getting-started.md:226]: so much better!

**Review by byjlw — APPROVED** [review]
I added a couple of comments, but overall this is way better and feel free to merge

File: docs/source/llm/getting-started.md [hunk 6]
```diff
@@ -14,221 +14,308 @@
...
-edge_config  =  EdgeCompileConfig(_check_ir_validity=False)
-edge_manager  =  to_edge(traced_model,  compile_config=edge_config)
-et_program  =  edge_manager.to_executorch()
+# Save the ExecuTorch program to a file.
+with open("nanogpt.pte", "wb") as file:
+    file.write(et_program.buffer)
 ```
 
-Also ExecuTorch provides different backend support for mobile acceleration. Simply call `to_backend()` with the specific backend partitioner on edge_manager  during exportation. Take Xnnpack delegation as an example:
+To export, run the script with `python export.py` (or python3, as appropriate for your environment). It will generate a `nanogpt.pte` file in the current directory.
```
Comment (kirklandsign) [inline docs/source/llm/getting-started.md:152]: Should be `export_nanogpt.py`?

File: docs/source/llm/getting-started.md [hunk 7]
```diff
@@ -14,221 +14,308 @@
...
 
-edge_config = edge_config = get_xnnpack_edge_compile_config()
-edge_manager = to_edge(traced_model, compile_config=edge_config)
-edge_manager = edge_manager.to_backend(XnnpackPartitioner())
+ExecuTorch provides a set of runtime APIs and types to load and run models.
 
-et_program = edge_manager.to_executorch()
-```
+```cpp
+// main.cpp
```
Comment (kirklandsign) [inline docs/source/llm/getting-started.md:162]: Do we plan to put them to the same piece?

File: docs/source/llm/getting-started.md [hunk 8]
```diff
@@ -318,24 +457,24 @@ m = convert_pt2e(m, fold_quantize=False)
...
-
+edge_manager = edge_manager.to_backend(XnnpackPartitioner()) # Lower to XNNPACK.
 et_program = edge_manager.to_executorch()
 ```
-Finally, update the CMakeLists.txt to link the XNNPACK backend with the runner.
+
+Finally, ensure that the runner links against the `xnnpack_backend` target in CMakeLists.txt.
 
 ```
 add_executable(nanogpt_runner nanogpt_runner.cpp)
```
Comment (kirklandsign) [inline docs/source/llm/getting-started.md:480]: is it main.cpp?

File: docs/source/llm/getting-started.md [hunk 9]
```diff
@@ -14,221 +14,308 @@
...
-
 ```
 
+Finally, download the following files into the same directory as main.h:
 
-And in the main function, we leverage the function to generate the outputs.
+TODO: This is a placeholder. Finish.
 ```
-vector<int64_t> outputs = generate(llm_model, tokens, sampler, /*target_output_length*/20);
+https://github.com/GregoryComer/et-tutorials/blob/quantization/nanogpt/managed_tensor.h
```
Comment (kirklandsign) [inline docs/source/llm/getting-started.md:291]: Use something like https://raw.githubusercontent.com/GregoryComer/et-tutorials/quantization/nanogpt/managed_tensor.h

File: docs/source/llm/getting-started.md [hunk 10]
```diff
@@ -348,15 +487,19 @@ target_link_libraries(
     xnnpack_backend) # Link the XNNPACK backend
```
Comment (kirklandsign) [inline docs/source/llm/getting-started.md:487]: Let's tell user exactly what's the diff?

Comment (kirklandsign) [reply docs/source/llm/getting-started.md:487]: Also etdump is causing issue?

File: docs/source/llm/getting-started.md [hunk 11]
```diff
@@ -348,15 +487,19 @@ target_link_libraries(
...
 
-1.  Get high level information
-get_delegation_info gives you a summary of what happened to the model after the to_backend() call:
+### Visualizing the Delegation
 
-```
+The `get_delegation_info()` method provides a summary of what happened to the model after the `to_backend()` call:
+
+```python
 from executorch.exir.backend.utils import get_delegation_info
```
Comment (kirklandsign) [inline docs/source/llm/getting-started.md:503]: which file?

File: docs/source/llm/getting-started.md (hunk 8 above)
Comment (GregoryComer) [reply docs/source/llm/getting-started.md:489]: Thanks. I updated the text here to be more clear.

File: docs/source/llm/getting-started.md (hunk 11 above)
Comment (GregoryComer) [reply docs/source/llm/getting-started.md:512]: Good catch. I'll update.

**Review by Gasoonjia — APPROVED** [review]

**Pull request closed without merging**